server/expression/in_subquery.go

```diff
@@ -97,7 +99,9 @@ func (in *InSubquery) Eval(ctx *sql.Context, row sql.Row) (any, error) {
 
 	// TODO: it might be possible for the left value to hash to a different value than the right even though they pass
 	//  an equality check. We need to perform a type conversion here to catch this case.
-	key, err := sql.HashOf(ctx, sql.NewRow(left))
+	// TODO: type conversions for numeric types can panic.
+	// TODO: this should be rType to match properly
```

Comment:
Rather than saying this should be the right type in the TODO, I'd change it to say that `hash.HashOf` expects the right type, which doesn't quite align with how we handle type casting.